app/src/components/avatar.rs

```diff
@@ -0,0 +1,133 @@
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "queries/get_user_avatar.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+pub struct GetUserAvatar;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct AvatarData(String);
+
+impl AvatarData {
+    pub fn new(data: String) -> Self {
+        AvatarData(data)
+    }
+}
+
+pub struct ShowAvatar {
+    common: CommonComponentParts<Self>,
+    avatar: Option<AvatarData>,
+    _producer: Box<dyn Bridge<AvatarEventBus>>,
+}
+
+/// State machine describing the possible transitions of the component state.
+/// It starts out by fetching the user's details from the backend when loading.
+pub enum Msg {
+    /// Received the user details response, either the user data or an error.
+    UserAvatarResponse(Result<get_user_avatar::ResponseData>),
+    Update(Response),
+}
+
+#[derive(yew::Properties, Clone, PartialEq, Eq)]
+pub struct Props {
+    pub username: String,
+    pub width: i32,
+    pub height: i32,
+}
+
+impl CommonComponent<ShowAvatar> for ShowAvatar {
+    fn handle_msg(
+        &mut self,
+        ctx: &Context<Self>,
+        msg: <Self as Component>::Message,
+    ) -> Result<bool> {
+        match msg {
+            Msg::UserAvatarResponse(response) => match response {
+                Ok(user) => self.avatar = user.user.avatar.map(AvatarData::new),
+                Err(e) => {
+                    self.avatar = None;
+                    bail!("Error getting user avatar: {}", e);
+                }
+            },
+            Msg::Update(Response::Update((username, avatar))) => {
+                if username == ctx.props().username {
+                    self.avatar = avatar;
+                    return Ok(true);
+                }
+                return Ok(false);
+            }
+        }
+        Ok(true)
+    }
+
+    fn mut_common(&mut self) -> &mut CommonComponentParts<Self> {
+        &mut self.common
+    }
+}
+
+impl ShowAvatar {
+    fn get_user_avatar(&mut self, ctx: &Context<Self>) {
+        self.common.call_graphql::<GetUserAvatar, _>(
+            ctx,
+            get_user_avatar::Variables {
+                id: ctx.props().username.clone(),
+            },
+            Msg::UserAvatarResponse,
+            "Error trying to fetch user avatar",
+        )
+    }
+}
+
+impl Component for ShowAvatar {
+    type Message = Msg;
+    type Properties = Props;
+
+    fn create(ctx: &Context<Self>) -> Self {
+        let mut avatar = Self {
+            common: CommonComponentParts::<Self>::create(),
+            avatar: None,
+            _producer: AvatarEventBus::bridge(ctx.link().callback(Msg::Update)),
+        };
+        avatar.get_user_avatar(ctx);
+        avatar
+    }
+
+    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+        CommonComponentParts::<Self>::update(self, ctx, msg)
+    }
+
+    fn view(&self, ctx: &Context<Self>) -> Html {
```

Comment:
This looks like a good candidate for the suspense API by yew: show a placeholder until the data is loaded. Maybe it would simplify things?